scenario/context.py

```diff
@@ -323,15 +323,15 @@ def pebble_custom_notice(container: Container, notice: Notice):
         )
 
     @staticmethod
-    def pebble_check_failed(container: Container, check: Check):
+    def pebble_check_failed(container: Container, check: CheckInfo):
```

Comment:
It's probably not a big deal as both arguments are position, but should this be called `info` or `check_info` for consistency? Similar with `pebble_check_recovered`.

Reply:
I agree with this - I think `info` is nicer since "check" is already there in the method name, and it's basically providing the content that will come back from the `event.info` call.

However, with this renaming done, the container collection name is a bit odd. For example:

```python
check = CheckInfo("http-check", failures=3, status=pebble.CheckStatus.DOWN)
container = Container("foo", checks={check})
state_in = State(containers={container})
state_out = ctx.run(ctx.on.pebble_check_failed(container, check), state_in)
```

And worse if I use keyword args:

```python
check = CheckInfo("http-check", failures=3, status=pebble.CheckStatus.DOWN)
container = Container("foo", checks={check})
state_in = State(containers={container})
state_out = ctx.run(ctx.on.pebble_check_failed(info=check, container=container), state_in)
```

`Container.checks` definitely can't be `Container.infos`. Maybe it has to be `Container.check_infos`? I don't really like "infos" as a plural, and I like the shorter name. It is most consistent with the other types, though. Thoughts?

Reply:
Yeah, it is slightly weird. But I'd probably go with `check_infos` though, for principle of least surprise (and I think "checks" is going to be confused with `pebble.Check`, that is, the plan/config).

```python
check_info = CheckInfo("http-check", failures=3, status=pebble.CheckStatus.DOWN)
container = Container("foo", checks_infos={check_info})
state_in = State(containers={container})
state_out = ctx.run(ctx.on.pebble_check_failed(info=check_info, container=container), state_in)
```

Speaking of naming, should `Container.service_status` be `service_statuses`?

Reply:
> Speaking of naming, should `Container.service_status` be `service_statuses`?

Yes, that's already part of https://github.com/canonical/ops-scenario/pull/124